include/deal.II/numerics/vector_tools.h

```diff
@@ -924,6 +928,26 @@ namespace VectorTools
                 VectorType &vec_result);
 
   /**
+   * Generic implementation for the project() function on parallel Triangulations with
+   * FiniteElements that are supported by the MatrixFree class.
+   * Use this function if you use more than four components or the degree of your FiniteElement
+   * is higher than eight. You have to specify the first template arguments yourself.
+   * @p vec_result is expected to not have any ghost entries.
+   */
+  template <int components, int fe_degree, int dim, typename VectorType, int spacedim>
+  void project_generic (const Mapping<dim, spacedim>                              &mapping,
```

Comment:
But even in that case we would still be left with the question whether we want users to be able to use the faster path, which requires to either look into `vector_tools.templates.h` or expose them in the header file. Since adding manual instantiations with `vector_tools.templates.h` is tricky (I don't know how many of our users are doing that), we might also just refer to the implementation in the documentation in the header file. I'm not sure what is best...